src/UnitfulAstro.jl

```diff
@@ -86,9 +86,76 @@ import UnitfulAngles: arcminute, arcsecond
 # Total electron content unit (used for ionospheric physics and low-frequency radio astronomy)
 @unit TECU       "TECU"     TotalElectronContentUnit  1e16*m^-2                 false
 
+@logscale mag "mag" Magnitude 10 -2.5 false
+
+#https://en.wikipedia.org/wiki/AB_magnitude     
+@logunit    AB_mag  "AB mag"        Magnitude   3631Jy
+
+# Bolometric magnitudes (cf IAU 2015)
+# https://en.wikipedia.org/wiki/Absolute_magnitude#Bolometric_magnitude
+@logunit    bol_Mag "bol Mag"       Magnitude   3.0128e28*W
+@logunit    bol_mag "bol mag"       Magnitude   2.518_021_002e-8*W*m^-2
+
+#https://en.wikipedia.org/wiki/Apparent_magnitude#Standard_reference_values
+#https://www.cfa.harvard.edu/~dfabricant/huchra/ay145/mags.html
+@logunit    U_mag   "Johnson U mag" Magnitude   1810Jy
+@logunit    B_mag   "Johnson B mag" Magnitude   4260Jy
+@logunit    V_mag   "Johnson V mag" Magnitude   3640Jy
+@logunit    R_mag   "Johnson R mag" Magnitude   3080Jy
+@logunit    I_mag   "Johnson I mag" Magnitude   2550Jy
+@logunit    J_mag   "Johnson J mag" Magnitude   1600Jy
+@logunit    H_mag   "Johnson H mag" Magnitude   1080Jy
+@logunit    K_mag   "Johnson K mag" Magnitude    670Jy
+@logunit    g_mag   "Gunn g mag"    Magnitude   3730Jy
+@logunit    r_mag   "Gunn r mag"    Magnitude   4490Jy
+@logunit    i_mag   "Gunn i mag"    Magnitude   4760Jy
+@logunit    z_mag   "Gunn z mag"    Magnitude   4810Jy
+
+Unitful.isrootpower_dim(::typeof(Unitful.dimension(AB_mag))) = false
+
+
+struct MagnitudeError <: Exception
+    x
+end
+Base.showerror(io::IO, e::MagnitudeError) = print(io, "MagnitudeError: $(e.x)")
+
+# https://en.wikipedia.org/wiki/Color_index
+# for color indices (subtracting magnitudes of different passbands)
+import Base.-
```

Comment:
All these methods are acts of [type piracy](https://docs.julialang.org/en/v1/manual/style-guide/index.html#Avoid-type-piracy-1), they definitely don't belong to this package, they should be moved to `Unitful.jl`

Reply:
This was inspired by [Unitful.jl itself and how they handle affine quantites](https://github.com/ajkeller34/Unitful.jl/blob/master/src/quantities.jl#L90)
```jl
# Disallow subtracting an affine quantity from a quantity
-(x::AbstractQuantity, y::AffineQuantity) =
    throw(AffineError("an invalid operation was attempted with affine quantities: $x - $y"))
```

Reply:
Sure, but these methods don't belong to this package.  They should be defined in `Unitful`, as they refer to `Base` functions with types defined only in that package, not here.

Reply:
These do need to be here (or we move all the  astronomy magnitude business to `Unitful.jl`).  The methods dispatch on `Magnitude`.  The method that allows for color indices to be computed:
```jl
import Base.-
function -(x::Unitful.Level{L,S,T},y::Unitful.Level{L,R,T}) where {L<:Magnitude,S,R,T<:Unitful.Quantity}
    return Unitful.ustrip(x) - Unitful.ustrip(y)
end
```
dispatches only on `Magnitude` log types.  Also notice the `S` and `R`, this permits only different magnitudes to be subtracted (hence the color index).  The other methods here are to allow for magnitudes to be added and subtracted from themselves and to keep people from trying to add differing magnitudes together (you would only subtract different magnitude bands, never add).

Reply:
Ok, I missed that `Magnitude` is defined here :grimacing:  So this doesn't make these method type piracy.

> (or we move all the astronomy magnitude business to `Unitful.jl`)

As you have probably seen in the last days, Michael has been a great supporter of adding this feature this to `Unitful.jl` as logarithmic units are heavily used in astronomy.

Still, I wonder if these methods wouldn't be a better fit in `Unitful.jl`.  Do you expect units outside those defined here that wouldn't require these methods?

I don't expect this package to need to define the arithmetic of logarithmic quantities, but only to define the units.

Reply:
So the real kicker here is that astronomers often compute color indices.  For example, if we had too magnitudes `B` and `V` with `B=0.3` and `V=0.9` then we would say `B-V=-0.6`.  This is not what you get if you convert back into `Jy` do the arithmetic and back.  This is a very astronomy thing and I wouldn't expect it to apply anywhere else.

Reply:
Ok, fine :slightly_smiling_face: